### PR TITLE
PROD-1214 Redact cli, database, and redis info from CONFIG_KEY_ALLOWLIST

### PR DESCRIPTION
**Pull request by daveqnet**
Partially closes [PROD-1214](https://ethyca.atlassian.net/browse/PROD-1214)

### Description Of Changes

* Removes `cli`, `database`, and `redis` configuration information from the allowable values returned by a `GET api/v1/config` API request.

### Code Changes

* [x] See commit (very simple changes)

### Steps to Confirm

* [x] Perform an authorized `GET api/v1/config` request and ensure that the response contains no `cli`, `database`, or `redis` data.

### Pre-Merge Checklist

* [ ] All CI Pipelines Succeeded
* Documentation:
  * [ ] documentation complete, [PR opened in fidesdocs](https://github.com/ethyca/fidesdocs/pulls)
  * [ ] documentation [issue created in fidesdocs](https://github.com/ethyca/fidesdocs/issues/new/choose)
* [x] Issue Requirements are Met
* [ ] Relevant Follow-Up Issues Created
* [x] Update `CHANGELOG.md`
* [ ] For API changes, the [Postman collection](https://github.com/ethyca/fides/blob/main/docs/fides/docs/development/postman/Fides.postman_collection.json) has been updated

[PROD-1214]: https://ethyca.atlassian.net/browse/PROD-1214?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by daveqnet** [comment]
Local test results:
```
~/projects/github/ethyca/fides daveqnet/PRO…EY_ALLOWLIST ❯ curl -i -X 'GET' \
  'http://localhost:8080/api/v1/config' \
  -H 'accept: application/json' \
  -H 'Authorization: Bearer eyJhbGciOiJkaXIiLCJlbmMiOiJBMjU2R0NNIn0..n6S1PTsXZRUu-KdMfTVi8Q.ONYluR86w927o1MLmoWTt_EUcIrupVThapcheGYyEjfj_tT5TpYPYOZmPgdxXgvxM-2xc-eEt-lLfo9fGkjcXMSlQM3x7gd7m9OH2qzliVgR_WxYKGlTmqqwOf4vks1TFbnRgevkt89pCKp4gSXU1_SKVLnREQ-8imIze7EEouEST3BpKTaGHhn3kK2QiNe0rIgWgnMHrK1z-lKnMlUo7W_YNo-pLFkYGk-fEVdLqzd3J7mlsulVnrS3FPBfJpJWnKo6jZGSb1m4E6C70i7VXOywb1D-SFdS8MbXyboP_xnJeBHaJpB9NyxmDXmR0dnLfcYnBhdpjFxpmklKkNHkAir_LCS3nex1z4YAMGQuVnvy-iu55OE93idVXgT7NrPu2MJIOdfblsBupUwpXraxyyTdcSGg7KNcAP7l0qwss-Th-HVsQcLGpBTXLj2ka7Pb3M4K8vJwngmXunVtmXPbFzY84UCcJOWGyVMUJEf8_k3MUyArQ3C4Mr7Ac59Ph0i-VslmGiPpDAmF-IhADy7zCJFKihndtQw5HbpZEIHH9x3f_G3Y9p8obLVldYo01gvHC-2u4RYo_O2RrorFTY7Eno68k6tdtpNdnpeONWvrjrPjy8kclR_aMnvvbLLMfaOtEJIbw7X5Wplq19N5Df5LZ05LWRG00hqYdIdPKrp22gCHGwtzYuwfFLuQ55XzINMKCHfnzXKj3rY8hNGS-_mN1CRJDuih2M4w8cuhmgLOXwPGfAXqwODNVTh99ZqG8CXKLnV1i9i7ADhSvRoYUWIyUD4Ii9oG7IPhwuN2Gbx3ES8ADpv6e9XozPqN8CtOEXTx_gZXyU00QDvtecPm2P1VbiqHNod_3KDyY1xzZz10l-xx93SvxWW5ZSs2JHJ-Mh0qA57aAghY9LhZNUyQiMgLVsftg4a-j-3tql9bpWRO_8OPWHhdZzJtM05OTF9esixPfeoSHOwnIuOVpCyGnxz2EMP-5sn3tsnmO-YGXtxkwApW5X9M9PiNbM4YW7moBimgFn_ioK3RsSYJsoH7cb_6rCQRn46MCk9ZiBz2M0yGp5GJ4JrjaF2eSVC3-YVDMnUJwJ0PO8oFHzsZKBjfJbxPNl3WIqhZh0SV2Xld1YJ0bdj8gTbqIbg5iVRFqQVijBItQvNnguwqKgu_4BLzAu751k3AhcTVcUXSrgFZjcHnEpbLiSAxxrkmx8b2cOo7u_b58BUUTaUi_9HpWEHdHyZ49dnRRsg0Zs53HO_b30Cgh0zBIsdgChnHybXovXzdE_-5i_GYbDlpuE6JeI_vrmxdcSCc7bLK7KZ6tCYHWZcJZ6ZLMrwVZLV744eMN4E1ijqz4xMyI_YbRwueRyv_GCY3nYiXFXaXDhZVaPr0NmEjtAnZI0TecO3vo43ElWmN_P59VSpjSvrJp0uJvEGECNSi13wXVC5sR0IgzGTS68z-BfQ9ln_OK1Ok6Oh9lfxmbhJDIdq3-CR_rTSVOd2hjLRxTyc6B2QyXBFKHKEtLs9hj70SqxaTrcXcu-bfWPG_KuKdFv_YzMKCBpUGf_1IKJTDEJji0vG4d2Tg7tD2YeeiRUMJeDtBOw_Z482WUrn-ERPjpa6PbUKIAEqjjFI3BpYiQEKo0ijJQQZCTMPEoFE90POo6Dqn2DaLqJfGECOXzJivMkLzWvde7wFM7kWeBXUQPdLoiz0NA8qkGgFrvy8hAKBFBSH_LCKiD1dO1MX114KOTB_Ht9WpAmzV55-WIQIwkRP3E786gZHm9lELQ3MTDnWgIPeGV5sTy_Ix8-ROTIv7VHVOWW_ZJ06uJl3PD-_mvrlBmGhpmKdE0XTv5OWvw0Ror8lhGvKNVQSd2NIaAOpIJTbMQ6VwxdB44QJ4hggAnN6P5NEwWC0LWaatcnFuLlz_Zp4EQ_Tiok__n-9Ols0e1D1RBR6oQzRY_fWNEZVIkXX0Sc2zbJpmOY8MVAiYSnYbKyp5zpmdxRSHtGgHLl5K80mQbl6FGlaNb7o6-XiVwC3Mg4InLnu-sX8N18YpuXQQr1t1dSwdzmZLooQ1ToXoWcMxoZuuDVHftPYg_a85Y3FCe2jEHGkqw_ovehsE14zpt81FoFR7zDoZEbkIKk2llH2lMpTqGVMytckYqf_AdmshgYf4fMnAZqDtSJ5Qs6CMhYz5AKoDkBlwDjj-exQJy98uEF7oBkgXAC5biOiujvDg_luswVD6GYP8SDMKTjoRsqp4WuDjYILFz02xnHMYwMOEjw1Elc3zXtQQJXAnjB8fvJTU42YRy4kX48gitmacOFyZkLaAjxnEjCBg7ZCCcr6GhwcsFzYvJ-TnpDVP-69hh_63EzkOXTJy26c7VtN0PvddlkD2RvyyA7OwCUKaTfhxhBQydAaKfVJRs4pyRJHYqZPPJ-XRICcj_4W1ZlgC_vC-dpQ5kHQq3JtW8JBt-Yp1AmRbn1vChOdooGkuk0PMu3NMOQ1Y9zG-tyUS3EthfreEcsGkNwJNMYZk16n5yD2ySl-1twOEgrDNtKFncrkT8krsdXJGj28QDTI7IHf4H_vbH69z9crETXHF1RHdba3ISRkQFIWtKcrZuBVWs1bfMz1-tKKFtybkC0HN5_b9H2boPcJqAzX5uZoDuLjQVUmKLYOmSZrJK6P9ooc0LnhcRIoSzdZ8jgoEBi86znOHbomPV_g8ENGf6mhb9qnFgTfwGqE7aedl6r9xJnIEXSwNxhNAdNuGsf1Sp35sqXyVHRzn5V8w0tvwOSIU_rISmw4KWxkzF30hVtyd23Ppxw3T3jFJ_mevP_9hmBczY4DSqRVlRLjvMCsqhipBsAnHp2hg3082KQUUVF5Dj6lm_VorW5_yEu4-6MziLsCE2zfOFokr4mIJOfzEaW8_acsQ53R5qsU6YirgoAWpptTPnUNvh8zRJeaFv5cpeEBTy-84Tz1SdZHfQDj_ubISdtpc2raYeHQnz53z32ajhtzBSat-OyQqMHKwDbjIl_N1eatMkb-6FghZiURILkCOmG2wrKBtT1_2M5MyG0PpWPKsSKonhOCJI36T6IuLPIGTZJp-KazYoZXpxMUBob-6XJ9qT9AC74OSBjNc9bGlN_C5UUFQitNCosvDyjYW93ChYY7A22br_YBVYbq9xAgPGPvWQvRDjT3-2jax4UPkUwRqRUUPmoKBXEa2O9yd6YZz77MOhGWoV6oyGtZ-qzJwnM0aWqMbnOI0ZPPmsM5Ktyv8YVVCZf-jw_TVfaNYqqwIb6ffPlDvrjOppaB4JElZpnL4eYNwpKda05hR9n5cFaxolrF9B1VTFiaDlOn167X-GoHxgBmp2HnFaq1NDbVM7zhQSP5nv9UF5K83Vm85255MoNRlb-D2RS6qW2yMQ7rfdusy2etCEI7Ens_vOuME_Ezu8iRKqxYh3SwbJ2gQdfFCKsBKRGIa4a7dle75xcbmi4HI6AxdctttwU9wp8yQkRxG2TByoRF1vlR8McjovoGBBAVpoKQw6D2cpxWe1l7n5bu8g6eunntaC5HHIOVnGnI1QAUla6Vp02gf5KwWf1yaOa2Reh-gGwL80y_pbhHBAYoTnHNuK2t7clIfEVvjVRNRsI5wC7kwSqRGvA-HTC5XKbM0UiZQAjT3HyBx0-GOLPtQchPq8bT7sxAIU5ab8AjtqvAfhnzlNRzPt3LqHSjEY7kGoLDfxTr7PwnzI_IuogUk3u0sIUN5rwDSEdi_OY1WTwYPZ6K2xK8H8DrczYEg794uPVvj0yFQkHcGXYLoHuuBbN4HKYYb0c63Sc1QVxRj_6izaEXx.H3U-VJisusI_ct7OsK8sQw'
```
```http
HTTP/1.1 200 OK
date: Fri, 03 Nov 2023 15:49:52 GMT
server: uvicorn
content-length: 571
content-type: application/json
x-ratelimit-limit: 1000
x-ratelimit-remaining: 998
x-ratelimit-reset: 1699026629
retry-after: Fri, 03 Nov 2023 15:50:29 -0000
```
```json
{
    "user":
    {
        "analytics_opt_out": false
    },
    "logging":
    {
        "level": "DEBUG"
    },
    "notifications":
    {
        "send_request_completion_notification": false,
        "send_request_receipt_notification": false,
        "send_request_review_notification": false,
        "notification_service_type": null
    },
    "security":
    {
        "cors_origins":
        [
            "http://localhost:8080",
            "http://localhost:3001",
            "http://localhost:3000"
        ],
        "encoding": "UTF-8",
        "oauth_access_token_expire_minutes": 11520
    },
    "execution":
    {
        "task_retry_count": 0,
        "task_retry_delay": 1,
        "task_retry_backoff": 1,
        "require_manual_request_approval": true,
        "subject_identity_verification_required": false
    }
}
```

**Comment by adamsachs** [comment]
thank you for being proactive and getting this change in @daveqnet! it looks good to me, just needs a quick update to the failing test which i'll get up right now.

i don't think we're relying on these values being returned via the API in any way, but i'll confirm quickly with the team that's the case just to not pull the rug out from under anyone without notice 👍

**Review by adamsachs — APPROVED** [review]
ok - updated test and confirmed it's passing locally, so i'm going to approve this then!

let's wait for CI to run green and it'll also give some time for folks to chime in with any objections. assuming that all goes well, then let's merge this in a bit! if any strong objections come in after the merge, we should always be able to revert before an official release anyway 👍

**Pull request merged**
